cmd/xl-v1-object.go

```diff
@@ -198,8 +198,14 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64, length i
 		return traceError(InvalidRange{startOffset, length, xlMeta.Stat.Size})
 	}
 
+	// Calculate endOffset according to length
```

Comment:
It is a trade off to avoid hitting the disk unnecessarily when client didn't want data in the first place. What needs to be sent to client is already available with GetObjectInfo. 

That is what it means by 'offset=0' and 'length=0' , value of validating quorum exists only if we need to serve data - when we are not going to serve anything it doesn't make sense to validate quorum. 0bytes file itself is quite meaningless on object storage - we kept it for compatibility reasons alone. 

I don't see any negative impacts of doing this.